tree-sitter/src/TreeSitter/Deserialize.hs

```diff
@@ -28,7 +28,7 @@ data Datatype
   = SumType
   { datatypeName       :: DatatypeName
   , datatypeNameStatus :: Named
-  , datatypeSubtypes   :: [Type]
+  , datatypeSubtypes   :: NonEmpty Type
```

Comment:
`fieldTypesToNestedSum` takes a `NonEmpty` list, and we want to make sure that the type synonym isn't empty.